src/ipc/ipc4/handler.c

```diff
@@ -1002,7 +1002,7 @@ void ipc_msg_reply(struct sof_ipc_reply *reply)
 
 void ipc_cmd(struct ipc_cmd_hdr *_hdr)
 {
-	struct ipc4_message_request *in = ipc_from_hdr(_hdr);
+	struct ipc4_message_request *in = ipc_from_hdr(&msg_data.msg_in);
```

Comment:
Minor note: the explanation is excellent in commit message, but ideally I'd also add a small oneliner comment in code. I fear in coming years, multiple developers will be scratching their heads looking at this function, and wondering why do we ignore "_hdr". Thankfully git-blame will help them locate the explanation, so I think this is anyways good.

Reply:
comment added